app/src/main/java/com/fivegmag/a5gmsmediastreamhandler/MediaSessionHandlerAdapter.kt

```diff
@@ -248,15 +249,19 @@ class MediaSessionHandlerAdapter() {
         val msg: Message = Message.obtain(null, SessionHandlerMessageTypes.CONSUMPTION_REPORTING_MESSAGE)
 
         // Generate fake reporting data
+        val ipv4Addr: String? = getIPAddress(4)
+        val ipv6Addr: String? = getIPAddress(6)
```

Comment:
Thanks @rjb1000 .
In 6.4.3.9 & 11.3.3.2, it says they're the IPs of the endpoint(UE?). So I think it should be client's IP. 
The getIPAddress() returns client's IP.

Did I misunderstand?

Reply:
> In 6.4.3.9 & 11.3.3.2, it says they're the IPs of the endpoint(UE?). So I think it should be client's IP.
>
> The getIPAddress() returns client's IP.

Yes, it's the 5GMS AS endpoint that is intended in TS 26.512.
* The _EndpointAddress_ type defined in clause 6.4.3.9 is intentionally neutral. It can be used to describe an endpoint at either end of a connection.
* The usage of this generic data type is defined in clause 11.3.3.2, so that's the important part. Re-reading the definition, I have realised that it is ambiguous and could be interpreted as either the client endpoint or the server endpoint.

> The IP address and port number of the endpoint used to access the media consumed.

I have raised 5G-MAG/Standards#91 to try and get clarity from SA4 in the next few weeks.